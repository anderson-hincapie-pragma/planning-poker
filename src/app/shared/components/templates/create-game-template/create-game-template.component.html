<div class="t-create-game">
    <app-input-card class="t-create-game__input-card"
        [labelText]="labelText"
        [inputType]="inputType"
        [inputPlaceholder]="inputPlaceholder"
        [inputControl]="inputControl"
        [actionButtonText]="actionButtonText"
        [buttonDisabled]="buttonDisabled"
        (actionButtonClick)="actionButtonClick.emit($event)">
    </app-input-card>
</div>