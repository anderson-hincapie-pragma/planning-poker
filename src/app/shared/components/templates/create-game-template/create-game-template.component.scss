:host {
    display: contents;
}
.t-create-game {
    width: 100vw;
    height: 100vh;
    background: radial-gradient(50% 50% at 50% 50%, #330072 0%, rgba(31, 13, 63, 0.7) 100%);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}