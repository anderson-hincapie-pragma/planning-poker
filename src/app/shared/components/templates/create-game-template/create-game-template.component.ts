import { Component, EventEmitter, Input, Output } from '@angular/core';
import { FormControl } from '@angular/forms';

@Component({
  selector: 'app-create-game-template',
  templateUrl: './create-game-template.component.html',
  styleUrls: ['./create-game-template.component.scss']
})
export class CreateGameTemplateComponent {

  @Input()
  public labelText: string = 'Label';

  @Input()
  public inputType: 'email' | 'number' | 'search' | 'text' = 'text';

  @Input()
  public inputPlaceholder: string = '';

  @Input()
  public inputControl: FormControl = new FormControl();

  @Input()
  public actionButtonText: string = 'Text';

  @Input()
  public buttonDisabled: boolean = false;

  @Output()
  public actionButtonClick: EventEmitter<MouseEvent> = new EventEmitter();

}
