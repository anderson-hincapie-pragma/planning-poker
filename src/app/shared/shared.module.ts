import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ActionButtonComponent } from './components/atoms/action-button/action-button.component';
import { InputComponent } from './components/atoms/input/input.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { FormInputComponent } from './components/molecules/form-input/form-input.component';
import { InputCardComponent } from './components/organisms/input-card/input-card.component';
import { CreateGameTemplateComponent } from './components/templates/create-game-template/create-game-template.component';



@NgModule({
  declarations: [
    ActionButtonComponent,
    InputComponent,
    FormInputComponent,
    InputCardComponent,
    CreateGameTemplateComponent
  ],
  imports: [
    CommonModule,
    ReactiveFormsModule
  ],
  exports: [
    ActionButtonComponent,
    InputComponent,
    FormInputComponent,
    InputCardComponent,
    CreateGameTemplateComponent
  ]
})
export class SharedModule { }
