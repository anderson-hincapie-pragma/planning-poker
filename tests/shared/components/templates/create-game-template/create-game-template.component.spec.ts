import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CreateGameTemplateComponent } from '@shared/components/templates/create-game-template/create-game-template.component';
import { SharedModule } from '@shared/shared.module';

describe('CreateGameTemplateComponent', () => {
  let component: CreateGameTemplateComponent;
  let fixture: ComponentFixture<CreateGameTemplateComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [CreateGameTemplateComponent]
    });
    fixture = TestBed.createComponent(CreateGameTemplateComponent);
    compiled = fixture.nativeElement;
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  })

});
