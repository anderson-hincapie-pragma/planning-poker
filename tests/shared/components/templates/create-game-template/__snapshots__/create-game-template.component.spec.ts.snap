// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGameTemplateComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="t-create-game"
  >
    <app-input-card
      class="t-create-game__input-card"
    >
      <div
        class="o-input-card"
      >
        <app-form-input
          class="o-input-card__form-input"
        >
          <div
            class="m-form-input"
          >
            <label
              class="m-form-input__label"
            >
              Label
            </label>
            <app-input
              class="m-form-input__input"
            >
              <input
                class="a-input ng-untouched ng-pristine ng-valid"
                placeholder=""
                type="text"
              />
            </app-input>
          </div>
        </app-form-input>
        <app-action-button
          class="o-input-card__action-button"
        >
          <button
            class="a-action-button"
            data-testid="action-button"
          >
            Text
          </button>
        </app-action-button>
      </div>
    </app-input-card>
  </div>
</div>
`;
